### ISSUE TITLE
[java/cs] Super method calls on the constructor

### ISSUE THREAD

**Issue by waneck** [HaxeFoundation/haxe#2805]
``` haxe
class Base { function test() {} }
class Child extends Base
{
    function new()
    {
        super.test();
    }
    override function test() { trace("child"); }
}
```

Fails on both Java and C#. Not sure how to deal with this.

**Comment by vroad** [comment]
Instead of converting entire constructor to a static method, can't we create static constructor that calls one instance method instead?

**Comment by theJenix** [comment]
I ran into this today in C#, and I'm kind of interested to take a look at the part of the Haxe compiler that generates this code.  Are you considering potential solutions here at this time, and if so can you point me in the right direction to get started?

**Comment by RealyUniqueName** [comment]
> Are you considering potential solutions here at this time

Yes, please.
You can probably start from here: https://github.com/HaxeFoundation/haxe/blob/bf252aa/src/codegen/gencommon/overloadingConstructor.ml#L26

**Comment by Simn** [comment]
The Java and C# targets will be put in "regression fixes only" state with the Haxe 4.3.0 release. See https://github.com/HaxeFoundation/haxe/wiki/Target-Support-Level#java-and-c-since-430 for additional information. We apologize for any inconvenience this causes!
